src/components/ProjectListItem/index.jsx

```diff
@@ -0,0 +1,91 @@
+import cls from 'classnames';
+import { useState, useEffect } from 'react';
+import ProjectStatusIcon from '../ProjectStatusIcon';
+import styles from './styles.module.css';
+import store from '../../store';
+import {
+  borders,
+  effects,
+  flex,
+  layout,
+  padding,
+  margin,
+  colors,
+  typography as type,
+} from '../../theme';
+
+export default function ProjectListItem({
+  last,
+  index,
+  projectSlug,
+  archiveProject,
+  isArchive,
+  selectedProject,
+  setSelectedProject,
+}) {
+  const [projectDetail, setProjectDetail] = useState();
+  const [status, setStatus] = useState(undefined);
+  const [projectName, setProjectName] = useState('...');
+
+  useEffect(() => {
+    (async () => {
+      if (!isArchive && projectSlug) {
+        const project = await store.getProject(projectSlug);
+        setProjectDetail(project);
+      } else if (isArchive && archiveProject) {
+        // when archive, project slug comes over as an object
+        setProjectName(archiveProject.name);
+      }
+    })();
+  });
+
+  useEffect(() => {
+    if (projectDetail) {
+      const { isUploaded, isPublished, name } = projectDetail;
+      setProjectName(name);
+      if (isPublished) {
+        setStatus('published');
+      } else if (isUploaded) {
+        setStatus('uploaded');
+      }
+    } else if (isArchive) {
+      setStatus('archive');
+    }
+    return () => {
+      setStatus(undefined);
+    };
+  }, [isArchive, projectDetail]);
```

Comment:
Here, you're using useState and useEffect when really I think this calls for a useMemo. The project details should be coming in via props from the ProjectList (since that's where the async call is) and then you can just determine things like status and project name from that in a useMemo.